Completion/Unix/Command/_find

```diff
@@ -37,6 +37,8 @@ case $variant in
     args+=(
       '*-Bmin:birth time (minutes)'
       '*-Bnewer:file to compare (birth time):_files'
+      '*-newer'{a,B,c,m}{a,B,c,m}'[succeed if timestamp 1 is newer than timestamp 2 of the given file]:reference file:_files'
```

Comment:
It's not clear to me what "timestamp 1" and "timestamp 2" are.  Instead, perhaps:

```suggestion
      '*-newer'{a,B,c,m}{a,B,c,m}'[if [aBcm\]time is newer than [aBcm\]time of given file]:reference file:_files'
```

Could probably do even better than this.

Reply:
The best help text would require separate entries for each combination of flags.  Or code generating such separate entries.

Reply:
The potential problem with that is that it'd use up 16  screen lines, and vertical space isn't that cheap.  (Because the output is single column… it'd be nice to change that, if $COLUMNS is large enough, but that's a separate discussion.)

Reply:
> [...] and vertical space isn't that cheap. (Because the output is single column… it'd be nice to change that, if $COLUMNS is large enough, but that's a separate discussion.)

I use [zsh4humans](https://github.com/romkatv/zsh4humans) which replaces the completion results with a tmux popup window with fzf search of the possible completions. I expect there are at least 10 other ways for users to completely replace how the completions are presented given that this is zsh.

Where am I going with this? Two points: 1) It is easy enough for users to change this if they want 2) Users *do* change things, so that limits what you can expect and assume anyway.

Reply:
> Where am I going with this? Two points: 1) It is easy enough for users to change this if they want 2) Users _do_ change things, so that limits what you can expect and assume anyway.

No. Please do not justify bad default behavior by asserting that "users can just change it if they want".

Reply:
> No. Please do not justify bad default behavior by asserting that "users can just change it if they want".

You are absolutely right, when you look at it from that point of view.

Reply:
IIRC, in this specific case the one-per-line behaviour is hardwired into `compdescribe`, the builtin [= implemented in C] guts of `_describe`, which `_arguments` calls.  (See uses of `zterm_columns` in [`bin_compdescribe()`](https://github.com/zsh-users/zsh/blob/master/Src/Zle/computil.c).)  This isn't something users can easily change, even if we assume `bin_compdescribe()` doesn't access zle's C internals in ways that a shell function can't.

zsh4humans and friends actually install a `compadd` wrapper function, which is problematic for reasons of composability (what if several plugins wanted to do that for different reasons?) and forward compatibility.

What the API _should_ be like is `zstyle :completion:settings:column-width 80` and then the screen would be divided into as many list-of-options columns as fit the terminal, each list-of-options column being 80 terminal columns wide.  Or perhaps `zstyle :completion:settings:preferred-completion-chooser myfunc` and then myfunc would get everything that's normally dumped to the terminal (group descriptions, candidate strings, candidate string descriptions; and all the forgoing for every element of tag-order) and would feed an answer back to compsys.